app/session.server.ts

```diff
@@ -36,15 +38,30 @@ export async function getUsername(request: Request): Promise<User['username'] |
   return username;
 }
 
+export async function getEffectiveUsername(request: Request): Promise<string> {
+  const cookie = request.headers.get('Cookie');
+  const effectiveUsername = await effectiveUsernameCookie.parse(cookie);
+  return effectiveUsername;
```

Comment:
Let's return this like so:

```ts
return effictiveUsername as User['username'];
```

And you can drop the return type on the function.